test/framework/kubernetes.go

```diff
@@ -218,3 +237,73 @@ func CreateSecret(namespace, name string, secretContent map[string]string) error
 
 	return kubernetes.ResourceC(kubeClient).Create(secret)
 }
+
+// GetIngressURI returns the ingress URI
+func GetIngressURI(namespace, serviceName string) (string, error) {
+	ingress, err := WaitForIngress(namespace, serviceName)
+	if err != nil || ingress == nil || len(ingress.Spec.Rules) == 0 {
+		return "", fmt.Errorf("Ingress %s does not exist in namespace %s: %v", serviceName, namespace, err)
+	}
+
+	return fmt.Sprintf("http://%s:80", ingress.Spec.Rules[0].Host), nil
+}
+
+// WaitForIngress waits for ingress by a service
+func WaitForIngress(namespace, serviceName string) (*v1beta1.Ingress, error) {
+	ingress := &v1beta1.Ingress{}
+	if exists, err :=
+		kubernetes.ResourceC(kubeClient).FetchWithKey(types.NamespacedName{Name: serviceName, Namespace: namespace}, ingress); err != nil {
+		return nil, err
+	} else if !exists {
+		return nil, nil
+	} else {
+		return ingress, nil
+	}
+}
+
+// ExposeServiceOnKubernetes adds ingress CR to expose a service
+func ExposeServiceOnKubernetes(namespace, serviceName string) error {
+	host := serviceName + namespace
+	if config.IsLocalTests() {
```

Comment:
This is useful for local tests as we can modify our /etc/hosts file to map the minikube ip to a service host. 
In non local, the final host will be something like service-name + namespace, so we avoid conflicts when running parallel executions.